.github/workflows/publish-apidocs.yml

```diff
@@ -0,0 +1,143 @@
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - include/onnxruntime/core/session
+      - csharp
+      - java
+      - onnxruntime/python
+  workflow_dispatch:
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCFXVERSION: 2.62.2
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: C API docs - install doxygen and dependencies
+        run: |
+          sudo apt update
+          sudo apt-get install libclang-dev
+          sudo apt-get install libclang-cpp14
+          wget https://www.doxygen.nl/files/doxygen-1.9.6.linux.bin.tar.gz
+          tar xvzf doxygen-1.9.6.linux.bin.tar.gz
+
+      - name: C API docs - run doxygen
+        run: |
+          mkdir -p build/doxygen
+          cd docs/c_cxx
+          ../../doxygen-1.9.6/bin/doxygen
+
+      - name: C# API docs - setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 5.0.x
+
+      - name: C# API docs - restore dependencies
+        run: dotnet restore csharp/ApiDocs/ApiDocs.csproj
+
+      - name: C# API docs - download tools
+        run: |
+          mkdir -p build/docfx
+          wget https://github.com/dotnet/docfx/releases/download/v${DOCFXVERSION}/docfx-linux-x64-v${DOCFXVERSION}.zip -O build/docfx/docfx.zip
+          unzip build/docfx/docfx.zip -d build/docfx
+
+      - name: C# API docs - install NuGet
+        uses: nuget/setup-nuget@v1
+
+      - name: C# API docs - build
+        run: |
+          build/docfx/docfx metadata csharp/ApiDocs/docfx.json
+          dotnet build csharp/ApiDocs/ApiDocs.csproj --no-restore
+          build/docfx/docfx build csharp/ApiDocs/docfx.json
+
+      - name: Java API docs - set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+
+      - name: Java API docs - build
+        uses: gradle/gradle-build-action@v2
+        with:
+          build-root-directory: java
+          gradle-executable: java/gradlew
+          arguments: javadoc
+
+      - name: Python API docs - install tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install python3-pip
+          sudo apt-get install graphviz
+      - name: Python API docs - install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          cd docs/python
+          python3 -m pip install -r requirements.txt
+          python3 -m pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ ort-nightly
```

Comment:
why do we need to install the ORT nightly package? is the current source code not sufficient to generate the Python docs?

Reply:
That's a good question. I'll look into it (but after we get this PR merged)